File: matlab-pipeline/NBSR_TIP12/NBSR_core/obs_for_SR.m
function y = obs_for_SR(x, k,  sigma, zoom)
% forward observation model
% Haichao Zhang
% 2011-8-6 19:27:
r = zoom;

%y = conv2(x, k/sigma,'valid');

y = real(ifft2(fft2(x).* psf2otf(k/sigma, size(x))));

y = y(1:r:end, 1:r:end);

File: python_pipeline/obs_for_SR.py
from pypher.pypher import psf2otf
import numpy as np

def obs_for_SR(x, k, sigma, zoom):
    y = np.real(np.fft.ifft2(np.multiply(np.fft.fft2(x), psf2otf(k/sigma, x.shape))))
    y = y[0::zoom, 0::zoom]

    return y